libs/config/src/analyze/lints/c15_cfgpatches_scope.rs

```diff
@@ -0,0 +1,309 @@
+        }
+    }
+    (patch_units, patch_weapons)
+}
+
+pub enum PatchType {
+    Vehicle,
+    Weapon,
+}
+impl PatchType {
+    const fn singular(&self) -> &str {
+        match self {
+            Self::Vehicle => "unit",
+            Self::Weapon => "weapon",
+        }
+    }
+    const fn base(&self) -> &str {
+        match self {
+            Self::Vehicle => "CfgVehicles",
+            Self::Weapon => "CfgWeapons",
+        }
+    }
+}
+
+macro_rules! define_cfgpatch_code {
+    ($name:ident, $ident:expr, $message_fn:expr, $label_fn:expr) => {
+        pub struct $name {
+            classname: String,
+            patch_type: PatchType,
+            span: Range<usize>,
+            severity: Severity,
+            diagnostic: Option<Diagnostic>,
+        }
+
+        impl Code for $name {
+            fn ident(&self) -> &'static str {
+                $ident
+            }
+            fn link(&self) -> Option<&str> {
+                Some("/lints/config.html#cfgpatches_scope")
+            }
+            fn severity(&self) -> Severity {
+                self.severity
+            }
+            fn message(&self) -> String {
+                $message_fn(&self.classname, &self.patch_type)
+            }
+            fn label_message(&self) -> String {
+                $label_fn(&self.patch_type)
+            }
+            fn diagnostic(&self) -> Option<Diagnostic> {
+                self.diagnostic.clone()
+            }
+        }
+
+        impl $name {
+            #[must_use]
+            pub fn new(
+                classname: String,
+                patch_type: PatchType,
+                span: Range<usize>,
+                processed: &Processed,
+                severity: Severity,
+            ) -> Self {
+                Self {
+                    classname,
+                    patch_type,
+                    span,
+                    severity,
+                    diagnostic: None,
+                }
+                .generate_processed(processed)
+            }
+            fn generate_processed(mut self, processed: &Processed) -> Self {
+                self.diagnostic =
+                    Diagnostic::from_code_processed(&self, self.span.clone(), processed);
+                self
+            }
+        }
+    };
+}
+
+define_cfgpatch_code!(
+    Code15CfgPatchItemNotFound,
+    "L-C15a",
+    |classname: &str, patch_type: &PatchType| format!(
+        "CfgPatches {}s[] class `{}` not found",
+        patch_type.singular(),
+        classname
+    ),
+    |patch_type: &PatchType| format!("not defined in {}", patch_type.base())
+);
+
+define_cfgpatch_code!(
+    Code15CfgPatchPublicItemNotListed,
+    "L-C15b",
+    |classname: &str, patch_type: &PatchType| format!(
+        "Public {} class `{}` not listed in CfgPatches",
+        patch_type.base(),
+        classname
+    ),
+    |patch_type: &PatchType| format!("has scope=2 but not in {}s[]", patch_type.singular())
```

Comment:
```suggestion
    |patch_type: &PatchType| format!("not in {}s[]", patch_type.singular())
```

I think it'd be better to add more info as a note, and probably add help messages with links to the relevant wiki pages.

Someone new to Arma modding will have no clue what these messages are about